### PR TITLE
Hardening/1751 syntax for uri param q and exists

### PR DESCRIPTION
**Pull request by kzangeli**
Changed +attr/-attr   for attr/!attr in q-string
Added lots of sanity checks for q and q-parts.

Fixes issues #1751 and #1754 
Apart from these two issues, q is inspected not to:
- be empty
- contain only a ';'
- contain two consecutive ';'s

These two last checks had to be made before the strtok-loop as strtok doesn't return empty tokens, it just eats them ...

### CODE REVIEW

File: test/functionalTest/cases/1592_query_with_q_op_greater_or_equal/query_with_all_q_ops_without_right_hand_side.test [hunk 1]
```diff
@@ -30,27 +30,35 @@ brokerStart CB
 --SHELL--
```
Comment (fgalan) [inline test/functionalTest/cases/1592_query_with_q_op_greater_or_equal/query_with_all_q_ops_without_right_hand_side.test]: (Discussed at skype) Add some .test to queck attribute existence and not-existence in query operation.

Comment (kzangeli) [inline test/functionalTest/cases/1592_query_with_q_op_greater_or_equal/query_with_all_q_ops_without_right_hand_side.test]: Functest added in 0b1d496

**Comment by fgalan** [comment]
LGTM (passing the ball to @crbrox for final LGTM + merge + delete branch)

**Comment by crbrox** [comment]
LGTM

**Pull request merged**